### PR TITLE
support miniBatch for WDL

### PR DESCRIPTION
**Pull request by wuhaifengdhu**
1, pass SE check for WDL
2, support miniBatch for WDL

When config miniBatchSize = 1000, the average iteration takes about 2s.
image

### CODE REVIEW

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java [hunk 1]
```diff
@@ -143,6 +165,17 @@ public WDLParams doCompute() {
...
     }
 
+    public WDLParams doCompute(int iteration, int trainBatchSize, int validateBatchSize) {
+        if(validateBatchSize < 10) {
+            LOG.info("Origin validationBatchSize is " + validateBatchSize + " which is so less and adjust to a large value");
+            validateBatchSize = Math.min(this.testData.size(), 10);
+        }
+        LOG.info("training on iteration: " + iteration + " with trainBatchSize: " + trainBatchSize + ", validateBatchSize:" + validateBatchSize);
+        adjustTrainSet(iteration * trainBatchSize % this.trainData.size(), trainBatchSize);
+        adjustTestSet(iteration * validateBatchSize % this.testData.size(), validateBatchSize);
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java:175]: Could we not enable mini batch in validation set? Validation set no need and better to leverage all validation set each time for a better validation error value.

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java [hunk 2]
```diff
@@ -143,6 +165,17 @@ public WDLParams doCompute() {
         return params;
     }
 
+    public WDLParams doCompute(int iteration, int trainBatchSize, int validateBatchSize) {
+        if(validateBatchSize < 10) {
+            LOG.info("Origin validationBatchSize is " + validateBatchSize + " which is so less and adjust to a large value");
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java:170]: Could it be warning or add verification in ModelProbe...?

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLWorker.java [hunk 3]
```diff
@@ -334,6 +334,10 @@ public void load(GuaguaWritableAdapter<LongWritable> currentKey, GuaguaWritableA
         this.updateMetrics(data, isInTraining);
     }
 
+    private boolean miniBatchEnabled() {
+        return !"1".equals(this.modelConfig.getTrain().getParams().getOrDefault(CommonConstants.MINI_BATCH, "1").toString());
+    }
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLWorker.java:339]: This condition is weird:)

Comment (wuhaifengdhu) [reply src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLWorker.java:339]: Removed compare with 1.

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java (hunk 2 above)
Comment (wuhaifengdhu) [reply src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java]: Add ModelProbe for WDL

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java (hunk 1 above)
Comment (wuhaifengdhu) [reply src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java]: Remove minibatch for validation set

File: src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java [hunk 4]
```diff
@@ -68,8 +68,22 @@ public WDLParallelGradient(final WideAndDeep wnd, int threadNumber, ConcurrentMa
         this.lossType = lossType;
 
         assert threadNumber > 0 && threadNumber < 33;
+        adjustTrainSet(0, this.trainData.size());
+        adjustTestSet(0, this.testData.size());
+    }
+
+    /**
+     * [start, end)
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java:76]: Could you comment this method well?

Comment (wuhaifengdhu) [reply src/main/java/ml/shifu/shifu/core/dtrain/wdl/WDLParallelGradient.java]: Added comments.

**Pull request merged**